lib/class-wp-theme-json-gutenberg.php

```diff
@@ -2361,10 +2362,13 @@ protected static function compute_style_properties( $styles, $settings = array()
 				$root_variable_duplicates[] = substr( $css_property, $root_style_length );
 			}
 
-			// Processes background styles.
-			if ( 'background' === $value_path[0] && isset( $styles['background'] ) ) {
-				$background_styles = gutenberg_style_engine_get_styles( array( 'background' => $styles['background'] ) );
-				$value             = $background_styles['declarations'][ $css_property ] ?? $value;
+			// Processes background image styles.
+			if ( 'background-image' === $css_property && ! empty( $value ) ) {
```

Comment:
We never really needed to pass all global styles background styles to the style engine because, aside from `backgroundImage`, none of them needed any processing. 

The style engine sanitizes the `backgroundImage`  image URI and wraps it in `url()`.

`$value` already contains resolved refs thanks to [$value = static::get_property_value](https://github.com/WordPress/gutenberg/pull/64128/files#diff-b03597cc3da199e5aa5a94950e8241135904853e7c3b82d758e42744878afae7R2353) above.

Reply:
This needs a bit of refactor now the following PR has been merged:

- https://github.com/WordPress/gutenberg/pull/64192

Reply:
Probably in a new PR so it's easier to test.